pkgs/development/libraries/g2o/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, stdenv, fetchFromGitHub, cmake, eigen, suitesparse }:
+
+stdenv.mkDerivation rec {
+  pname = "g2o";
+  version = "unstable-2019-04-07";
+
+  src = fetchFromGitHub {
+    owner = "RainerKuemmerle";
+    repo = pname;
+    rev = "9b41a4ea5ade8e1250b9c1b279f3a9c098811b5a";
+    sha256 = "1rgrz6zxiinrik3lgwgvsmlww1m2fnpjmvcx1mf62xi1s2ma5w2i";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ eigen suitesparse ];
+
+  meta = {
+    description = "A General Framework for Graph Optimization";
+    homepage = "https://github.com/RainerKuemmerle/g2o";
```

Comment:
https://github.com/RainerKuemmerle/g2o states that http://openslam.org/g2o.html is the homepage for the project, which states that the source code is available at https://github.com/OpenSLAM-org/openslam_g2o

@lopsided98 can you please explain your choice for `src` and `homepage`?

Reply:
The OpenSLAM-org repo is abandoned and has not been updated since 2013. The OpenSLAM page claims (near the bottom) that this is the stable release, and RainerKuemmerle's repo is the development version. This may have been true at one time, but there have been more than 1000 commits over 6 years to the new repo since the "stable release", so it doesn't really make sense to use the abandoned repo.

I also decided not to use the OpenSLAM page as the homepage because it contains outdated information and does not describe the software in its current state.

Reply:
Sounds reasonable to me. Thanks for clarifying!